### PR TITLE
Support FreeBSD in Cirrus CI

### LINKED ISSUE #9597: Support FreeBSD and macOS in Cirrus CI
Opening this issue to see if now that the migration to Cirrus CI has been settled, we can start supporting FreeBSD in our CI?
Also since macOS is supported, if we have plans of supporting it as well.

/cc @fertapric

### PR DESCRIPTION
**Pull request by eksperimental**
Closes #9597

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -114,3 +114,41 @@ test_windows_task:
...
   test_elixir_script: make --keep-going test_elixir
+
+
+test_freebsd_task:
+  name: FreeBSD 12.0
+
+  freebsd_instance:
+    image_family: freebsd-12-0
+    cpu: 8
+    memory: 7424Mi
```
Comment (fertapric) [inline .cirrus.yml]: What's your process to get these memory values? (just curious)

Comment (eksperimental) [reply .cirrus.yml]: they are given by the CI. If you go bellow, it will raise an error telling what is the minimum and the maximum for that many CPUs.

Comment (josevalim) [reply .cirrus.yml]: So do we need to set them at all?

Comment (eksperimental) [reply .cirrus.yml]: I wouldn't be able to tell what's the default value being used
https://cirrus-ci.com/task/5629539488628736

I prefer to go with as little memory as needed, so we can get a slot as quick as possible.

Comment (eksperimental) [reply .cirrus.yml]: > So do we need to set them at all?

@josevalim yes, we do

no cpu , no memory set (default values defined by Cirrus= cpu: 2, memory: 4GB)
finished in 8min 30sec
https://cirrus-ci.com/task/6362284892356608

cpu: 8, it errors.
https://cirrus-ci.com/task/4845801095823360
Failed to start: Invalid value for field 'resource.machineType': 'https://www.googleapis.com/compute/v1/projects/cirrus-ci-community/zones/us-central1-f/machineTypes/custom-8-4096'. Memory size for 8 vCPU instance should be between 7424MiB and 53248MiB, while 4096MiB is requested


cpu: 8, memory: 7424Mi
finished in 6min
https://cirrus-ci.com/task/6338292030636032

cpu: 8, memory: 14848Mi
finished in 6min
https://cirrus-ci.com/task/6343139740090368


So the best option is the original values of cpu: 8, memory: 7424Mi

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Fantastic job @eksperimental! It is very appreciated!

**Comment by eksperimental** [comment]
@josevalim No worries. Always a pleasure to contribute to Elixir.

**Pull request merged**